security/secure-markup.md

```diff
@@ -0,0 +1,52 @@
+# Secure Markup
+
+TODO: add TOC
+
+## Scope of this document
+Developers should be aware of the security implications of their front-end code. This document outlines some steps developers can take to reduce the risk of markup exposing us, or our users, to security flaws.
+
+Note: this document does not cover JavaScript, or interactions which may touch the server, e.g. XSS or HTTP headers. (It is assumed security-related headers are set in the HTTP headers, not in `meta` tag equivalents.)
+
+Additionally, developers are encouraged to familiarise themsevles with the [OWASP HTML5 Security Cheat Sheet](https://www.owasp.org/index.php/HTML5_Security_Cheat_Sheet).
+
+## Use UTF-8 and specify this in a `meta` tag
+
+Do this: `<meta charset="utf-8">`.
+
+Not specifying the appropriate character set has historically has been a source of [charset-related security problems](https://code.google.com/archive/p/doctype-mirror/wikis/ArticleUtf7.wiki) as well as a source of rendering bugs.
+
+## Add `rel="noopener"` to outbound links in new windows
+
+Links leak the context of the opening window in the `window.opener` object. This allows the opened window to alter the opening window via JavaScript, regardless of whether the window is opened via JavaScript or simply a `target="_blank"` attribute. Exploiting this context leaking is known as ["tabnabbing"](https://mathiasbynens.github.io/rel-noopener/).
+
+## Use sane form defaults
+
+Sanitising user input is the job of the server and/or JavaScript, but we can help by specifying some form widget attributes as part of a ["belt & braces"](https://www.collinsdictionary.com/dictionary/english/belt-and-braces) approach to security:
+
+**TODO: how contentious is ^that?**
+
+- `form` &mdash; specify `accept-charset` and `enctype` **TODO: not specifying these smells bad (I seem to remember bugs around this), but is it?**
+- `input` &mdash; if the value of the type attribute is `text`, `email`, `search`, `password`, `tel`, or `url`, specify the `maxlength` attribute.
+- `textarea` &mdash; specify the `maxlength` attribute.
```

Comment:
Do not check just the length, but expected character set too.  A whitelist approach is recommended.

Reply:
Cool, should go on the `form` element. IIRC some widgets can also specify a charset and/or enctype in HTML _5_ will check ;)